cmd/thanos/compact.go

```diff
@@ -189,6 +188,11 @@ func runCompact(
 		return errors.Wrap(err, "create compactor")
 	}
 
+	readinessProber, err := metricHTTPListenGroup(g, logger, reg, httpBindAddr, component.Compact)
```

Comment:
This really confuses me.. `create readiness prober` but actually function `metricHTTPListenGroup` name does not suggest that? Can we rename `metricHTTPListenGroup` to something meaningful?

Reply:
Not resolved at all

Reply:
Sorry, I removed initialization of the prober outside of this function so here it just registers and thought this way it does not change the semantics of the function so the name can stay the same. I should have express myself here better. 

Not sure if the new name should be more generic or on contrary explicit.
`defaultHTTPListener` or `defaultMetricsReadinessListener` ? 
(I'll go with the `defaultHTTPListener` for now but I'll be glad if you say what suits sou best)

Reply:
renamed it here https://github.com/improbable-eng/thanos/pull/656/commits/4810ac1cbf0de4c4197bacce2bc4b8fada33ccde

PTAL it that is ok with you or if I should name it otherwise.